internal/host/plugin/repository_host_catalog.go

```diff
@@ -250,8 +250,6 @@ func (r *Repository) UpdateCatalog(ctx context.Context, c *HostCatalog, version
 			// field, as secrets do not have a database entry. Clear the
 			// secrets out of the working set after.
 			newCatalog.Secrets = c.Secrets
-			c.Secrets = nil
```

Comment:
Why the change? Just curious. Also the comment should probably be updated.

Reply:
In the case where we have 2 secret fields we are updating this code gets executed twice.  The first time assigns value to newCatalog.Secrets and nil's out c.Secrets, the second time through it nils out newCatalog.Secrets since we nil'ed out c.Secrets in the first pass.

Reply:
Updated comment.

Reply:
Huh, ok -- I thought we were only doing wholesale updates of secrets.

Reply:
We are but the field mask is still populated with all the subfields of secrets.

Reply:
Ah! That's what I was missing. Thanks!